game_patch/graphics/d3d11/gr_d3d11_hooks.cpp

```diff
@@ -264,7 +264,7 @@ namespace df::gr::d3d11
     static CodeInjection vif_lod_mesh_dtor_injection{
         0x005695D0,
         [](auto& regs) {
-            rf::VifLodMesh* lod_mesh = regs.ecx;
+            const auto lod_mesh = addr_as_ref<rf::VifLodMesh*>(regs.esp + 4);
```

Comment:
This code accesses data from the stack frame of the caller function. It is not argument for this function and can be anything depending on what calls this function. It is unacceptable.
What is wrong with ecx (this) argument? This code runs at the beginning of the destructor before object is deallocated, so how is it use-after-free?